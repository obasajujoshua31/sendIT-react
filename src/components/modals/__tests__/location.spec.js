import React from "react";
import { shallow } from "enzyme";
import sinon from "sinon";
import {
  mapDispatchToProps,
  mapStateToProps,
  ChangeLocation
} from "../location";

const props = {
  parcelId: "1",
  role: false,
  history: {
    push: jest.fn()
  },
  show: true,
  presentLocation: "joshua",
  editLocation: jest.fn(),
  destination: "dkkdkd",
  parcelStatus: false,
  closeChangeLocation: jest.fn()
};
const mockStore = {
  parcels: {
    parcelStatus: true
  }
};
const mockSubmit = {
  preventDefault: jest.fn()
};

const mockHandleChange = value => ({
  target: {
    value
  }
});

const wrapper = shallow(<ChangeLocation {...props} />);

describe("Edit ChangeLocation Test", () => {
  it("should render Edit ChangeLocation", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("dispatch should call loadOneParcel", () => {
    const dispatchSpy = sinon.spy();
    const dispatchResult = mapDispatchToProps(dispatchSpy);
    dispatchResult.editLocation("1", "Abuja");
    expect(dispatchSpy.callCount).toEqual(1);
  });

  it("map state to props should return parcels", () => {
    const mapStateResult = mapStateToProps(mockStore);
    expect(mapStateResult.parcelStatus).toEqual(true);
  });

  it("handleCloseModal should set setLocationModal state to false", () => {
    const handleCloseSpy = jest.spyOn(wrapper.instance(), "handleCloseModal");

    handleCloseSpy();
    expect(wrapper.instance().state.showLocationModal).toEqual(false);
  });

  it("changePresentLocation should not call props.editLocation wgeb there is error", () => {
    const changeLocationSpy = jest.spyOn(
      wrapper.instance(),
      "changePresentLocation"
    );
    wrapper.setState({
      formError: "kandoki"
    });
    changeLocationSpy(mockSubmit);
    expect(wrapper.instance().props.editLocation).not.toBeCalled();
  });

  it("changePresentLocation should call props.editLocation", () => {
    const changeLocationSpy = jest.spyOn(
      wrapper.instance(),
      "changePresentLocation"
    );
    wrapper.setState({
      presentLocation: "Abuja",
      formError: ""
    });
    changeLocationSpy(mockSubmit);
    expect(wrapper.instance().props.editLocation).toBeCalledWith("1", "Abuja");
  });

  it("handle change should set errors when e.target.value is not defined", () => {
    const handleChangeSpy = jest.spyOn(wrapper.instance(), "handleChange");

    handleChangeSpy(mockHandleChange(""));
    expect(wrapper.instance().state.formError).toEqual(
      "Present Location cannot be empty"
    );
  });

  it("handle change should set presentLocation when e.target.value is defined", () => {
    const handleChangeSpy = jest.spyOn(wrapper.instance(), "handleChange");

    handleChangeSpy(mockHandleChange("Abuja"));
    expect(wrapper.instance().state.presentLocation).toEqual("Abuja");
  });

  it("componentDidMount should set state to this.prop.presentLocation", () => {
    const componentSpy = jest.spyOn(wrapper.instance(), "componentDidMount");

    wrapper.setProps({
      presentLocation: "Abuja"
    });
    componentSpy();
    expect(wrapper.instance().state.presentLocation).toEqual("Abuja");
  });

  it("should call props.history.push when parcelStatus is true", () => {
    wrapper.setProps({
      parcelStatus: true
    });
    expect(wrapper.instance().props.history.push).toBeCalledWith(
      "/admin-dashboard"
    );
  });

  it("should render status modal if state is the same as props", () => {
    wrapper.setState({
      parcelStatus: false
    });
    wrapper.setProps({
      parcelStatus: false
    });
    expect(wrapper).toMatchSnapshot();
  });
});
