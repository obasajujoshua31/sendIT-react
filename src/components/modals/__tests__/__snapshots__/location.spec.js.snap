// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit ChangeLocation Test should render Edit ChangeLocation 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="editModal"
  closeTimeoutMS={0}
  contentLabel="Modal #1 Global Style Override Example"
  isOpen={true}
  onRequestClose={[MockFunction]}
  overlayClassName="Overlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    style={
      Object {
        "alignItem": "center",
        "backgroundColor": "#806B75",
        "display": "flex",
        "justifyContent": "center",
        "padding": 15,
        "position": "relative",
      }
    }
  >
    <span
      style={
        Object {
          "bottom": 5,
          "color": "white",
          "cursor": "pointer",
          "fontSize": "1.8em",
          "padding": 5,
          "position": "absolute",
          "right": 5,
        }
      }
    >
      <i
        className="far fa-times-circle"
        onClick={[MockFunction]}
      />
    </span>
    <h3
      style={
        Object {
          "color": "white",
          "fontSize": "1.4rem",
          "fontWeight": "bold",
        }
      }
    >
      Change Parcel Location
    </h3>
  </div>
  <div
    style={
      Object {
        "maxWidth": "100%",
        "padding": "10px",
      }
    }
  >
    <form
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <label
          style={
            Object {
              "color": "#806B75",
              "flex": 3,
              "fontWeight": "bold",
            }
          }
        >
          Present Location:
        </label>
        <input
          name="presentLocation"
          onChange={[Function]}
          style={
            Object {
              "borderColor": "rgba(0, 0, 0, 0.3)",
              "borderRadius": 10,
              "color": "#806B75",
              "flex": 4,
              "fontSize": "14px",
              "outline": "none",
              "padding": "8px 10px",
              "width": "100%",
            }
          }
          type="text"
          value="joshua"
        />
      </div>
      <button
        className="button edit-direction"
      >
        change Location
      </button>
    </form>
  </div>
</Modal>
`;

exports[`Edit ChangeLocation Test should render status modal if state is the same as props 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="editModal"
  closeTimeoutMS={0}
  contentLabel="Modal #1 Global Style Override Example"
  isOpen={true}
  onRequestClose={[MockFunction]}
  overlayClassName="Overlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    style={
      Object {
        "alignItem": "center",
        "backgroundColor": "#806B75",
        "display": "flex",
        "justifyContent": "center",
        "padding": 15,
        "position": "relative",
      }
    }
  >
    <span
      style={
        Object {
          "bottom": 5,
          "color": "white",
          "cursor": "pointer",
          "fontSize": "1.8em",
          "padding": 5,
          "position": "absolute",
          "right": 5,
        }
      }
    >
      <i
        className="far fa-times-circle"
        onClick={[MockFunction]}
      />
    </span>
    <h3
      style={
        Object {
          "color": "white",
          "fontSize": "1.4rem",
          "fontWeight": "bold",
        }
      }
    >
      Change Parcel Location
    </h3>
  </div>
  <div
    style={
      Object {
        "maxWidth": "100%",
        "padding": "10px",
      }
    }
  >
    <form
      noValidate={true}
      onSubmit={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "preventDefault": [MockFunction] {
                  "calls": Array [
                    Array [],
                    Array [],
                  ],
                  "results": Array [
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                  ],
                },
              },
            ],
            Array [
              Object {
                "preventDefault": [MockFunction] {
                  "calls": Array [
                    Array [],
                    Array [],
                  ],
                  "results": Array [
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                  ],
                },
              },
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": null,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <label
          style={
            Object {
              "color": "#806B75",
              "flex": 3,
              "fontWeight": "bold",
            }
          }
        >
          Present Location:
        </label>
        <input
          name="presentLocation"
          onChange={
            [MockFunction] {
              "calls": Array [
                Array [
                  Object {
                    "target": Object {
                      "value": "",
                    },
                  },
                ],
                Array [
                  Object {
                    "target": Object {
                      "value": "Abuja",
                    },
                  },
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": <Modal
                    ariaHideApp={true}
                    bodyOpenClassName="ReactModal__Body--open"
                    className="editModal"
                    closeTimeoutMS={0}
                    contentLabel="Modal #1 Global Style Override Example"
                    isOpen={true}
                    onRequestClose={[MockFunction]}
                    overlayClassName="Overlay"
                    parentSelector={[Function]}
                    portalClassName="ReactModalPortal"
                    role="dialog"
                    shouldCloseOnEsc={true}
                    shouldCloseOnOverlayClick={true}
                    shouldFocusAfterRender={true}
                    shouldReturnFocusAfterClose={true}
                  >
                    <div
                      style={
                        Object {
                          "alignItem": "center",
                          "backgroundColor": "#806B75",
                          "display": "flex",
                          "justifyContent": "center",
                          "padding": 15,
                          "position": "relative",
                        }
                      }
                    >
                      <span
                        style={
                          Object {
                            "bottom": 5,
                            "color": "white",
                            "cursor": "pointer",
                            "fontSize": "1.8em",
                            "padding": 5,
                            "position": "absolute",
                            "right": 5,
                          }
                        }
                      >
                        <i
                          className="far fa-times-circle"
                          onClick={[MockFunction]}
                        />
                      </span>
                      <h3
                        style={
                          Object {
                            "color": "white",
                            "fontSize": "1.4rem",
                            "fontWeight": "bold",
                          }
                        }
                      >
                        Change Parcel Location
                      </h3>
                    </div>
                    <div
                      style={
                        Object {
                          "maxWidth": "100%",
                          "padding": "10px",
                        }
                      }
                    >
                      <form
                        noValidate={true}
                        onSubmit={
                          [MockFunction] {
                            "calls": Array [
                              Array [
                                Object {
                                  "preventDefault": [MockFunction] {
                                    "calls": Array [
                                      Array [],
                                      Array [],
                                    ],
                                    "results": Array [
                                      Object {
                                        "type": "return",
                                        "value": undefined,
                                      },
                                      Object {
                                        "type": "return",
                                        "value": undefined,
                                      },
                                    ],
                                  },
                                },
                              ],
                              Array [
                                Object {
                                  "preventDefault": [MockFunction] {
                                    "calls": Array [
                                      Array [],
                                      Array [],
                                    ],
                                    "results": Array [
                                      Object {
                                        "type": "return",
                                        "value": undefined,
                                      },
                                      Object {
                                        "type": "return",
                                        "value": undefined,
                                      },
                                    ],
                                  },
                                },
                              ],
                            ],
                            "results": Array [
                              Object {
                                "type": "return",
                                "value": null,
                              },
                              Object {
                                "type": "return",
                                "value": undefined,
                              },
                            ],
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "display": "flex",
                            }
                          }
                        >
                          <label
                            style={
                              Object {
                                "color": "#806B75",
                                "flex": 3,
                                "fontWeight": "bold",
                              }
                            }
                          >
                            Present Location:
                          </label>
                          <input
                            name="presentLocation"
                            onChange={
                              [MockFunction] {
                                "calls": Array [
                                  Array [
                                    Object {
                                      "target": Object {
                                        "value": "",
                                      },
                                    },
                                  ],
                                  Array [
                                    Object {
                                      "target": Object {
                                        "value": "Abuja",
                                      },
                                    },
                                  ],
                                ],
                                "results": [Circular],
                              }
                            }
                            style={
                              Object {
                                "borderColor": "rgba(0, 0, 0, 0.3)",
                                "borderRadius": 10,
                                "color": "#806B75",
                                "flex": 4,
                                "fontSize": "14px",
                                "outline": "none",
                                "padding": "8px 10px",
                                "width": "100%",
                              }
                            }
                            type="text"
                            value="Abuja"
                          />
                        </div>
                        <button
                          className="button edit-direction"
                        >
                          change Location
                        </button>
                      </form>
                    </div>
                  </Modal>,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            }
          }
          style={
            Object {
              "borderColor": "rgba(0, 0, 0, 0.3)",
              "borderRadius": 10,
              "color": "#806B75",
              "flex": 4,
              "fontSize": "14px",
              "outline": "none",
              "padding": "8px 10px",
              "width": "100%",
            }
          }
          type="text"
          value="Abuja"
        />
      </div>
      <button
        className="button edit-direction"
      >
        change Location
      </button>
    </form>
  </div>
</Modal>
`;
